src/__readlink_chk.c

```diff
@@ -41,7 +42,12 @@ wrapper(__readlink_chk, ssize_t, (const char * path, char * buf, size_t bufsiz,
     debug("__readlink_chk(\"%s\", &buf, %zd, %zd)", path, bufsiz, buflen);
     expand_chroot_path(path);
 
-    if ((linksize = nextcall(__readlink_chk)(path, tmp, FAKECHROOT_PATH_MAX-1, buflen)) == -1) {
+    if (__builtin_expect(!!(bufsiz > buflen), 0)) {
+      printf("readlink: prevented write past end of buffer\n");
```

Comment:
why the printf to stdout? at most there is a print to stderr done in the rest of the codebase